include/oneapi/dpl/internal/histogram_extension_defs.h

```diff
@@ -0,0 +1,47 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_HISTOGRAM_EXTENSION_DEFS_H
+#define _ONEDPL_HISTOGRAM_EXTENSION_DEFS_H
+
+#include "../pstl/onedpl_config.h"
+
+namespace oneapi
+{
+namespace dpl
+{
+
+// Only supported for dpcpp backend
+#ifdef _ONEDPL_BACKEND_SYCL
+
```

Comment:
@MikeDvorskiy @SergeyKopienko @mmichel11 @adamfidel and all:

@rarutyun and I discussed a possible change of the API to something like this:

```
template <typename _Size, typename _ValueType>
oneapi::dpl::even_histogram_config
make_histogram_config(_Size num_bins, _ValueType first_bin_min_val, _ValueType last_bin_max_val);

template <typename _RandomAccessIterator>
oneapi::dpl::custom_range_histogram_config
make_histogram_config(_RandomAccessIterator boundary_first, _RandomAccessIterator boundary_last);


template <typename _ExecutionPolicy, typename _RandomAccessIterator1, typename _HistogramConfig, typename _RandomAccessIterator2>
oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy, _RandomAccessIterator2>
histogram(_ExecutionPolicy&& exec, _RandomAccessIterator1 first, _RandomAccessIterator1 last, _HistogramConfig histogram_config,
          _RandomAccessIterator2 histogram_first);
```
where `_HistogramConfig` must be `even_histogram_config` or `custom_range_histogram_config`.

Calls could look like:
``` 
histogram(exec, input_first, input_last, make_histogram_config(10, 0, 100), output_first);

auto hist_config = make_histogram_config(boundary_first, boundary_last);

histogram(exec, input_first, input_last, hist_config, output_first);
```

We want to raise it to the group to see if there is any strong support for such a change of API.  A brief listing of pros and cons we see:

Pros:
* Allows fewer overloads for supporting a mixture of ranges and iterators (for potential future range APIs)
* Allows easier re-use of a histogram "configuration" with different inputs / outputs

Cons:
* Departure from other std parallel APIs, with addition of separated configuration 
* Makes migration more complicated within SYCLomatic
* Lose ability to connect the type of bin range min and max to the value type of the input sequence for evenly divided API

We have not found overwhelming motivation for this change, but I think it is worth raising as an option.  
If there is not strong support for this change, we will leave the API as it is in the PR currently.

Please share your thoughts.